cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -442,9 +440,10 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, String namespac
     Map<NamespacedEntityId, Metadata> systemMetadata = fetchMetadata(sortedEntities, MetadataScope.SYSTEM);
     Map<NamespacedEntityId, Metadata> userMetadata = fetchMetadata(sortedEntities, MetadataScope.USER);
 
+    Set<MetadataSearchResultRecord> resultRecords = addMetadataToEntities(sortedEntities, systemMetadata, userMetadata);
     return new MetadataSearchResponse(
-      sortInfo.getSortBy() + " " + sortInfo.getSortOrder(), offset, limit, numCursors, total,
-      addMetadataToEntities(sortedEntities, systemMetadata, userMetadata), cursors, showHidden
+      sortInfo.getSortBy() + " " + sortInfo.getSortOrder(), offset, limit, numCursors, resultRecords.size(),
```

Comment:
That's with this fix, right?
And without the fix, the total is much higher (depending on the offset provided).